### DISCUSSION TITLE
SortableList item margin has $backgroundDefault color?

### DISCUSSION THREAD

**Discussion by YuyouWu** [wix/react-native-ui-lib#2538] [Q&A] [answered]
Hi, 

I'm making a simple todo app with `SortableList`. When I apply margin to the item inside `renderItem`, the margin area will have a color of the $backgroundDefault color. 
![IMG_0309](https://user-images.githubusercontent.com/16201149/225814872-0d657473-f1cb-4444-ab51-db0fdf6763d3.png)
The code for `renderItem` is below. If I try to add `backgroundColor:"black"` to the `View` component, it will only make what's underneath the list item black. The margin area remains $backgroundDefault. 
```  
const renderItem = useCallback(
    ({ item, index: _index }: { item: Item; index: number }) => {
      return (
        <View
          margin-5
        >
          <Drawer
            rightItems={[
              {
                text: "Delete",
                background: "red",
              },
            ]}
            style={{
              borderRadius: 5,
            }}
          >
            <TouchableHighlight
              onPress={() => {
                console.log(item);
              }}
              style={{
                height: 62,
                backgroundColor: "#111111",
                borderRadius: 5,
              }}
            >
              <Text
                marginH-10
                color="white"
                style={{ marginTop: "auto", marginBottom: "auto" }}
              >
                {item.text}
              </Text>
            </TouchableHighlight>
          </Drawer>
        </View>
      );
    },
    []
  );
```

I am able to hide the color by using padding and setting background color to match its container's background color:
![IMG_0308](https://user-images.githubusercontent.com/16201149/225815414-b04191f3-8530-4c8e-90a2-5695256263b1.png)
```  
const renderItem = useCallback(
    ({ item, index: _index }: { item: Item; index: number }) => {
      return (
        <View
          paddingV-5
          paddingH-10
          style={{
            backgroundColor: "black",
          }}
        >
          <Drawer
            rightItems={[
              {
                text: "Delete",
                background: "red",
              },
            ]}
            style={{
              borderRadius: 5,
            }}
          >
            <TouchableHighlight
              onPress={() => {
                console.log(item);
              }}
              style={{
                height: 62,
                backgroundColor: "#111111",
                borderRadius: 5,
              }}
            >
              <Text
                marginH-10
                color="white"
                style={{ marginTop: "auto", marginBottom: "auto" }}
              >
                {item.text}
              </Text>
            </TouchableHighlight>
          </Drawer>
        </View>
      );
    },
    []
  );
```
but when dragging an item, you can see the margin is there (tile named `five`). 

Is there anyway to remove this color from margin? Is this intended behavior? 

Thanks in advance for any advice. 

Update:
Changing line 121 `backgroundColor: Colors.$backgroundDefault,` in `node_modules\react-native-ui-lib\src\components\sortableList\SortableListItem.js` to `backgroundColor: "transparent",` seems to resolve the issue:
![IMG_0310](https://user-images.githubusercontent.com/16201149/225818279-fdeb4381-db30-44d7-8265-d0241d01978d.png)

I'm not sure what elevation means but I've tested on Android also. Please advice.

**Comment by lidord-wix** [comment, marked as answer]
Hello @YuyouWu,
I tried to change this color to `'transparent'` and it causing some bugs: 
1. You can see here that the items are transparent when they overlap: 
![transparent](https://user-images.githubusercontent.com/66782556/232484912-7f05d091-e480-4fe7-a28a-d8102f5d21f7.png)
instead of:
![non-transparent](https://user-images.githubusercontent.com/66782556/232484932-1168689e-b6d3-4ebb-bebe-650e92985c11.png)
2. The `elevation` on Android adds a shadow effect to make the element elevated ("closer" to the user).
[Android-transparent.webm](https://user-images.githubusercontent.com/66782556/232485249-8b554ffb-3d05-472f-b7f2-73a09a504ebd.webm)
instead of: 
[Android-non-transparent.webm](https://user-images.githubusercontent.com/66782556/232485260-8fa121fe-ab21-4938-8c04-aba8abced6c3.webm)

Anyway, I found a solution for that with a new prop called `itemMargins`, please see https://github.com/wix/react-native-ui-lib/pull/2561 and check it with your case.
You can pass the prop directly to the SortableList component, for example: 
```
<SortableList
  ...
  itemMargins={{marginBottom: 20, marginLeft: 10, marginRight: 10}}
/>
```
You can also install react-native-ui-lib@7.3.0-snapshot.2948 to check that.
Please let me know if it works for you and we'll merge that :)

**Reply by YuyouWu** [reply]
Thanks for the response! I will comment on the PR.
